Frends.SFTP.DownloadFiles/Frends.SFTP.DownloadFiles/Definitions/FileTransporter.cs

```diff
@@ -4,459 +4,446 @@
+                return Encoding.ASCII;
+            case FileEncoding.ANSI:
+                return Encoding.Default;
+            case FileEncoding.Unicode:
+                return Encoding.Unicode;
+            case FileEncoding.WINDOWS1252:
+                return Encoding.Default;
+            case FileEncoding.Other:
+                return Encoding.GetEncoding(dest.FileNameEncodingInString);
+            default:
+                throw new ArgumentOutOfRangeException();
+        }
+    }
 
-            // return Tuple with empty list and success.true if files are not found.
-            if (files.Count() == 0)
-                return new Tuple<List<FileItem>, bool>(fileItems, true);
+    /// <summary>
+    /// Get source files that fit the file name / mask
+    /// </summary>
+    /// <param name="client"></param>
+    /// <param name="source"></param>
+    /// <returns></returns>
+    private Tuple<List<FileItem>, bool> GetSourceFiles(SftpClient client, Source source)
+    {
+        var fileItems = new List<FileItem>();
 
-            // create List of FileItems from found files.
-            foreach (var file in files)
-            {
-                if (Util.FileMatchesMask(Path.GetFileName(file.FullName), source.FileName))
-                {
-                    FileItem item = new FileItem(file);
-                    _logger.NotifyInformation(_batchContext, $"FILE LIST {item.FullPath}");
-                    fileItems.Add(item);
-                }
-                    
-            }
-            return new Tuple<List<FileItem>, bool>(fileItems, true);
+        if (_filePaths != null)
+        {
+            fileItems = _filePaths.Select(p => new FileItem(p) { Name = p }).ToList();
+            if (fileItems.Any()) return new Tuple<List<FileItem>, bool>(fileItems, true);
+            return new Tuple<List<FileItem>, bool>(fileItems, false);
         }
 
-        private static List<FileItem> GetDestinationFiles(string directory)
-        {
-            var destFiles = Directory.GetFiles(directory);
-            var result = new List<FileItem>();
-            foreach (var file in destFiles)
-            {
-                result.Add(new FileItem(file));
-            }
+        // Return empty list if source directory doesn't exists.
+        if (!client.Exists(SourceDirectoryWithMacrosExtended)) return new Tuple<List<FileItem>, bool>(fileItems, false);
 
-            return result;
-        }
+        // fetch all file names in given directory
+        var files = client.ListDirectory(SourceDirectoryWithMacrosExtended).ToList();
 
-        private static string[] ConvertObjectToStringArray(object objectArray)
-        {
-            var res = objectArray as object[];
-            return res?.OfType<string>().ToArray();
-        }
+        // return Tuple with empty list and success.true if files are not found.
+        if (files.Count() == 0) return new Tuple<List<FileItem>, bool>(fileItems, true);
```

Comment:
List has a property called Count, so you should use that instead of inherited method call. So use `files.Count` instead of `files.Count()`.

Reply:
Fixed.